app/services/salesforce_service/base.rb

```diff
@@ -1,31 +1,24 @@
 require 'restforce'
 require 'facets/hash/rekey'
-require "#{Rails.root}/lib/faraday_custom_headers.rb"
 
 module SalesforceService
   # encapsulate all Salesforce querying functions in one handy service
   class Base
     class_attribute :retries
-    class_attribute :headers
+    class_attribute :error
     self.retries = 1
 
-    Faraday::Middleware.register_middleware(
-      custom_headers: CustomHeaders,
-    )
-
     def self.client
       Restforce.new
     end
 
     def self.oauth_client
-      client = Restforce.new(
+      Restforce.new(
         authentication_retries: 0,
         oauth_token: oauth_token,
         instance_url: ENV['SALESFORCE_INSTANCE_URL'],
         mashify: false,
       )
-      client.middleware.use :custom_headers, headers if headers.present?
```

Comment:
Yup. This `custom_headers` thing I came up with didn't seem to properly send the headers through to salesforce 😞  so I got rid of it -- the `api_post_with_headers` function replaces it.